### PR TITLE
tuya_thermostat (TS0601): modes changed to presets and two other fixes

### PR DESCRIPTION
**Pull request by mgrom**
This PR is related to: https://github.com/Koenkk/zigbee2mqtt/pull/4070

- window detection fix - before there was no way to change this value through mqtt command,
- changed preset to away preset params - something named preset was actually away preset mode params (temp, and duration in days),
- modes changed to presets - modes in home assistant are fixed, so the best to be flexible is to use presets (ha calls them hold_modes)

### CODE REVIEW

**Comment by Koenkk** [comment]
I've just merged https://github.com/Koenkk/zigbee-herdsman-converters/pull/1461 which conflicts with this pr, can you rebase this pr?

Note to self: after this merge https://github.com/Koenkk/zigbee2mqtt/pull/4070

**Comment by mgrom** [comment]
No problem, I'll do it today.

**Comment by mgrom** [comment]
@Koenkk I'm only a bit worried about this change: 
https://github.com/Koenkk/zigbee-herdsman-converters/blob/470125e9b277eb4615b06cfc39dce83f76422391/converters/fromZigbee.js#L315
Because it can break currently working installations. As I mentioned in other PR, homeassistant is not allowing to add any custom system modes, so to be able to control mode we should use presets instead, which are more flexible. 
But this change can make some damage for user who currently use this integration without homeassistant.
Any ideas how to proceed?

**Comment by Koenkk** [comment]
Returning both is fine!

**Comment by mgrom** [comment]
I think it's ready @Koenkk

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -307,15 +308,16 @@ const tuyaThermostat = (model, msg, publish, options, meta) => {
...
-        return {preset: dataAsDecNumber};
-    case 1028: // 0x0404 Mode changed
-        if (utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatSystemMode').hasOwnProperty(dataAsDecNumber)) {
-            return {system_mode: utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatSystemMode')[dataAsDecNumber]};
+    case 626: // 0x7202 away preset temperature
+        return {away_preset_temperature: dataAsDecNumber};
+    case 629: // 0x7502 away preset number of days
+        return {away_preset_days: dataAsDecNumber};
+    case 1028: // 0x0404 Preset changed
+        if (utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatPreset').hasOwnProperty(dataAsDecNumber)) {
```
Comment (Koenkk) [inline converters/fromZigbee.js:316]: this changes breaks e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/eafa4a8dfc9298541827363b06c893297ccce9e2/devices.js#L1529

Comment (mgrom) [reply converters/fromZigbee.js]: I've forgot to check SYstemMode dictionary. Fixed

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**